### PR TITLE
Workspace dropdown menu

### PR DESCRIPTION
**Pull request by joeizang**

### CODE REVIEW

File: libs/remix-ui/workspace/src/lib/remix-ui-workspace.tsx [hunk 1]
```diff
@@ -292,91 +293,181 @@ export function Workspace () {
...
+        id='workspaceRename'
+        data-id='workspaceRename'
+        onClick={(e) => {
+          e.stopPropagation()
+          renameCurrentWorkspace()
+          _paq.push(['trackEvent', 'fileExplorer', 'workspaceMenu', 'workspaceRename'])
+          hideIconsMenu(!showIconsMenu)
+        }}
+        className='far fa-edit w-100'>
+          {'  Rename'}
```
Comment (LianaHus) [inline libs/remix-ui/workspace/src/lib/remix-ui-workspace.tsx:316]: IMHO it's better to style with css, For that please separate icon and the label into 2 separate spans and fix the spacing between them.

File: libs/remix-ui/workspace/src/lib/remix-ui-workspace.tsx [hunk 2]
```diff
@@ -292,91 +293,181 @@ export function Workspace () {
...
+                    className='far fa-plus-square remixui_menuicon'
+                    >
+                  </span>
+                  </OverlayTrigger>
+                  <Dropdown id="workspacesMenuDropdown" data-id="workspacesMenuDropdown" onToggle={() => hideIconsMenu(!showIconsMenu)} show={showIconsMenu}>
+                    <Dropdown.Toggle
+                      as={CustomIconsToggle}
+                      mouseOverAction={(e: SyntheticEvent) => {
+                        hideIconsMenu(!showIconsMenu)
+                        console.log({ e })
```
Comment (LianaHus) [inline libs/remix-ui/workspace/src/lib/remix-ui-workspace.tsx:452]: do we need this?

**Comment by LianaHus** [comment]
I wonder if click should also work on the menu. hover is fine, but i always try to click, and it doesn't work. Do you think we can have both?

File: libs/remix-ui/workspace/src/lib/remix-ui-workspace.tsx (hunk 1 above)
Comment (joeizang) [reply libs/remix-ui/workspace/src/lib/remix-ui-workspace.tsx:316]: Done. Used pl-2 rule for separation

File: libs/remix-ui/workspace/src/lib/remix-ui-workspace.tsx (hunk 2 above)
Comment (joeizang) [reply libs/remix-ui/workspace/src/lib/remix-ui-workspace.tsx:452]: No we don't. Left in error.

**Comment by yann300** [comment]
- the icon doesn't seem well placed and it isn't big enough. can we make all the icons a bit bigger?
 - the "create" should also be in the dropdown imo.
 - the dropdown view itself, I think it needs some more design. Something seems off but I don't know how to point out it...

**Comment by joeizang** [comment]
> I wonder if click should also work on the menu. hover is fine, but i always try to click, and it doesn't work. Do you think we can have both?

I will try to have both and see how it feels

**Comment by yann300** [comment]
- I would be even fine to remove the display on hover and keep the onClick @LianaHus

**Comment by joeizang** [comment]
> * I would be even fine to remove the display on hover and keep the onClick @LianaHus

I have moved this away from on hover to click now.

**Comment by yann300** [comment]
> the dropdown view itself, I think it needs some more design. Something seems off but I don't know how to point out it...

could you try:
 - add separation between "delete" and "download" and "restore" and "close"
 - make **all** the icons a bit bigger.
 - the onhover and blue background doesn't look.
 - the "plus" and "menu" are apart which is weird..

**Comment by ioedeveloper** [comment]
I agree with @yann300's comments. I also think tooltips background needs to change.

**Comment by yann300** [comment]
for the "clone" it should be the github icon

**Comment by yann300** [comment]
`clone` should be in a specific question

**Comment by yann300** [comment]
- The background when hovering a file is not right, it should be the same as for the file explorer.
 - open the dropdown, click on a file in the file explorer, this should hide the dropdown.

**Comment by yann300** [comment]
- can you increase the size of the Menu icon (settings `large` for font-size looks ok), and adapt with the `+` icon and the label "workspace"
 cc @ioedeveloper

**Comment by joeizang** [comment]
> `clone` should be in a specific question

I am sorry @yann300 I don't understand this comment, can you elaborate?

**Comment by yann300** [comment]
also "Workspace" should be all uppercase

**Comment by joeizang** [comment]
> also "Workspace" should be all uppercase

Done. Thanks @yann300

**Comment by Aniket-Engg** [comment]
Screenshot 2022-10-14 at 7 02 40 PM

here too

**Pull request merged**